src/com/ableton/VirtualEnv.groovy

```diff
@@ -15,12 +15,59 @@ class VirtualEnv implements Serializable {
    * Destination directory for the virtualenv. This value is set during construction of
    * the object, and is under the system temporary directory.
    */
-  String destDir
+  String destDir = null
+  /**
+   * Series of commands needed to install and active a version of Python inside the
+   * current shell.
+   */
+  String envSetup = null
+
+  protected String activateSubDir = null
+
+  /**
+   * Create a virtualenv using a specific version of Python, installed via pyenv. pyenv
+   * should be installed on the node, but the actual setup of any required environment
+   * variables (e.g. PYENV_ROOT and PATH) will be done inside this function.
+   *
+   * @param script Script context.
+   *               <strong>Required value, may not be null!</strong>
+   * @param python Python version, as given by pyenv versions --list.
+   * @param pyenvRoot Path to the installation of pyenv.
+   * @return New instance of VirtualEnv object.
+   */
+  static VirtualEnv create(Object script, String python, String pyenvRoot) {
+    if (!script.isUnix()) {
+      script.error 'This method is not supported on Windows'
+    }
 
-  protected String activateSubDir
+    VirtualEnv venv = new VirtualEnv(script, python)
+    assert pyenvRoot
+
+    if (!script.fileExists(pyenvRoot)) {
```

Comment:
I agree that this is nice, however @rco-ableton should be warned that this will break the tests and will require updating the `@Test(expected = ...)` parts, since we'll be raising only `Exception` now and not `AssertionError`.

Reply:
It would be nice but I think I'll leave this alone for now. Feel free to open a pull request updating it though!